includes/preprocessor_base.inc

```diff
@@ -107,20 +107,61 @@ abstract class IslandoraBatchPreprocessor {
   /**
    * Get an identifier we can use for an object.
    *
+   * @param IslandoraBatchObject $object
+   *   The object to be later processed and added to Fedora.
+   *
    * @return string
    *   An id to use in the back-end (Fedora).
    */
-  protected function getId() {
-    // Not specifying a namespace should result in one from the default
-    // declared in the fedora.fcfg ("changeme" by default).
-    $namespace = (isset($this->parameters['namespace'])) ?
-      $this->parameters['namespace'] :
-      NULL;
+  protected function getId(AbstractFedoraObject $object) {
+    // Command line argument overrides Collection Policy.
+    if (isset($this->parameters['namespace'])) {
+      $namespace = $this->parameters['namespace'];
+    }
+    else {
+      $namespace = $this->getNamespace($object);
+    }
 
     // TODO: Implement some form of caching, so we can get multiple at a time.
     return $this->connection->repository->api->m->getNextPid($namespace);
   }
 
+
+  /**
+   * Try to get the namespace from the collection policy.
+   *
+   * @param IslandoraBatchObject $object
```

Comment:
Sorry I saw the same thing at line 177, should the doc be updated or could we change the hint in the method?